### PR TITLE
Update tool versions; Improve logging

### PR DESCRIPTION
**Pull request by lpotthast**
Addresses https://github.com/ctron/trunk/issues/5

- Replaces `tracing::log::debug/info` with `tracing::debug/info`
- Some error message now including more information
- Tool installation now checks that the main executable exists, is a file and is actually executable
- We now cehck that the path to the style source exists and is a file before running the SASS tool
- Also updates tool versions
  -   Sass: "1.63.6" => "1.69.5"
  -   TailwindCss: "3.3.2" => "3.3.5"
  -   WasmBindgen: "0.2.87" => "0.2.88"
  -   WasmOpt: "version_113" => "version_116"

### CODE REVIEW

File: src/build.rs [hunk 1]
```diff
@@ -64,7 +64,12 @@ impl BuildSystem {
         // Ensure the output dist directories are in place.
         fs::create_dir_all(self.cfg.final_dist.as_path())
             .await
-            .with_context(|| "error creating build environment directory: dist")?;
+            .with_context(|| {
+                format!(
+                    "error creating build environment directory: {}",
+                    self.cfg.final_dist.to_string_lossy()
```
Comment (ctron) [inline src/build.rs:70]: I would go for `.display()` … but I'll fix this post merge.

Comment (lpotthast) [reply src/build.rs:70]: Can change this. Already did in another place. Btw, there are a few more occasions where `to_string_lossy()` is used when formatting.

File: src/tools.rs [hunk 2]
```diff
@@ -514,27 +543,39 @@ mod archive {
...
 
         Ok(out)
     }
 
     /// Set the executable flag for a file. Only has an effect on UNIX platforms.
-    fn set_file_permissions(file: &mut File, mode: u32) -> Result<()> {
+    fn set_file_permissions(
+        file: &mut File,
+        mode: u32,
+        file_path_hint: Option<impl AsRef<str>>,
```
Comment (ctron) [inline src/tools.rs:565]: I would also go for `impl Display` instead. That should be a better fit. But again, I can also do this post merge.

**Comment by ctron** [comment]
Many thanks for the PR! If you want to tackle the small things, please do. Otherwise I can also do them post-merge.

I did merge the other PR, merging in the upstream branch. Either that or one of the changed before cause some conflicts now in this PR.

**Review by ctron — APPROVED** [review]
Looks good in any case, waiting for the small fixes. Just ping me.

**Review by ctron — APPROVED** [review]

**Review by ctron — DISMISSED** [review]

**Review by ctron — APPROVED** [review]

**Comment by ctron** [comment]
I think you need to rebase the PR, looks like it's merged instead. Most likely that's the reason the CI doesn't start.

**Comment by lpotthast** [comment]
Changed a few `to_string_lossy` places and merged in the current trunk-ng state. History looks atrocious though. Squashing the commits would be great. I can open a PR with a squashed commit if this should not be possible again when merging.

**Comment by lpotthast** [comment]
Can you close this one in favour of https://github.com/ctron/trunk/pull/14?

**Pull request closed without merging**